lib/plugin/instance.rb

```diff
@@ -375,9 +375,14 @@ def register_group_param(param)
     DiscoursePluginRegistry.register_group_param(param, self)
   end
 
-  # Add a custom scopes for search to Group, respecting if the plugin is enabled
-  def register_group_scope_for_search(scope_name)
-    DiscoursePluginRegistry.register_group_scope_for_search(scope_name, self)
+  # Add a custom filter for search to Group
+  # Block takes groups and optional current_user
+  # For example:
+  # plugin.register_group_filter_for_search(:admins_filter) do |groups, user|
+  #   groups.where(name: "admins")
+  # end
+  def register_group_filter_for_search(filter, &block)
+    DiscoursePluginRegistry.group_filter_for_search[filter] = block
```

Comment:
I'm wondering if we should prevent the same filter from being registered twice here. Never know when two plugins might decide to use the same name so some protection here will be nice.